Microsoft.Azure.Cosmos/src/Query/DocumentQueryExecutionContextBase.cs

```
@@ -425,7 +425,21 @@ public DocumentServiceRequest CreateDocumentServiceRequest(INameValueCollection
             MemoryStream memoryStream = new MemoryStream();
             documentServiceResponse.ResponseBody.CopyTo(memoryStream);
             long responseLengthBytes = memoryStream.Length;
-            IJsonNavigator jsonNavigator = JsonNavigator.Create(memoryStream.ToArray());
+            byte[] content = memoryStream.ToArray();
+            IJsonNavigator jsonNavigator = null;
+
+            // Use the users custom navigator first. If it returns null back try the
+            // internal navigator.
+            if (this.feedOptions.CreateCustomNavigator != null)
+            {
+                jsonNavigator = this.feedOptions.CreateCustomNavigator(content);
+            }
+
+            if(jsonNavigator == null)
```

If the CreateCustomeNavigator call fails then their is a bug in the logic. The user should be setting the ContentSerializationFormat to something custom and then we use this navigator.

---

So we should just throw an exception instead.

---

If you still want to go for this then at least make the contract `TryCreateCustomNavigator` and the default implementation will always fail.

---

Fixing in next iteration